Index: HD.as
===================================================================
--- HD.as	(revision 219)
+++ HD.as	(working copy)
@@ -27,6 +27,7 @@
 	/** List with configuration settings. **/
 	public var config:Object = {
 		bitrate:1500,
+		autoswitch:false,
 		file:undefined,
 		fullscreen:false,
 		state:true
@@ -61,6 +62,12 @@
 		if(view.config['hd.file']) {
 			config['file'] = view.config['hd.file'];
 		}
+		if(view.config['hd.bitrate']) {
+			config['bitrate'] = view.config['hd.bitrate'];
+		}
+		if(view.config['hd.autoswitch']) {
+			config['autoswitch'] = view.config['hd.autoswitch'];
+		}
 		if(config['state'] == true) {
 			view.config['file'] = config['file'];
 		}
@@ -105,7 +112,7 @@
 		}
 		if(evt.data.bandwidth && !checked) {
 			checked = true;
-			if((evt.data.bandwidth < config['bitrate']) == config['state']) {
+			if(((evt.data.bandwidth < config['bitrate']) == config['state']) && (config['autoswitch'])) {
 				clickHandler();
 			}
 		}
